test/persistency.test.ts

```diff
@@ -45,6 +46,36 @@ test('persists cookies in localStorage', () => {
         'https://mswjs.io',
         [['cookieName', { name: 'cookieName', value: 'abc-123' }]],
       ],
-    ])
+    ]),
   )
 })
+
+describe('when `localStorage` is undefined', () => {
+  let originalLocalStorage: Storage
+
+  beforeEach(() => {
+    originalLocalStorage = localStorage
+
+    Object.defineProperty(window, 'localStorage', {
+      value: undefined,
+      writable: true,
+    })
+  })
+
+  afterEach(() => {
+    localStorage = originalLocalStorage
+  })
+
+  test('does not persists cookies', () => {
+    const req = new Request('https://mswjs.io')
+    const res = new Response(null, {
+      headers: new Headers({ 'set-cookie': 'cookieName=abc-123' }),
+    })
+
+    store.add(req, res)
+    store.persist()
+
+    localStorage = originalLocalStorage
+    expect(localStorage.getItem(PERSISTENCY_KEY)).toEqual(null)
```

Comment:
Despite the cookies not being persisted in the local storage, I think we should still assert that they persist in the internal `this.store` map when you try to retrieve them.

Reply:
Done @kettanaito